src/plugins/node/gc/nodegcplugin.cpp

```diff
@@ -149,7 +149,17 @@ void afterGC(v8::Isolate *isolate, GCType type, GCCallbackFlags flags) {
 	gcRealEnd = GetRealTime();
 
 	// GC type
-	const char *gcType = (type == kGCTypeMarkSweepCompact) ? "M" : "S";
+	const char *gcType = NULL;
+        switch (type) {
+          case kGCTypeMarkSweepCompact: gcType = "M"; break;
+          case kGCTypeScavenge: gcType = "S"; break;
+#if NODE_VERSION_AT_LEAST(5, 0, 0)
+          case kGCTypeIncrementalMarking: gcType = "I"; break;
+          case kGCTypeProcessWeakCallbacks: gcType = "W"; break;
+#endif
+          // Should never happen, but call it minor if type is unrecognized.
+          default: gcType = "S"; break;
```

Comment:
Can a GC always only be one type? The collector won't do two as one operation? The type values are designed to be masked together, that may be just for registration but I can't tell from the docs in `v8.h`.
As it stands if `type == kGCTypeMarkSweepCompact| kGCTypeProcessWeakCallbacks` the case statement would report gcType as `S` (`kGCTypeScavenge `).

Reply:
The pre-existing code here assumed it could only be one operation, as well as the new code in Node.js, but I'm looking through the v8 src now to confirm that is correct.

Reply:
Looking at the V8 code, it appears to me that the type is always a specific value, I think the code above is correct. @bnoordhuis you agree?

Reply:
Yes, V8 always reports them individually.

Reply:
Cool, I saw them being masked together and thought I'd better check.